File: models/FinancialDocument.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const FinancialDocument = sequelize.define('FinancialDocument', {
    documentType: {
      type: DataTypes.ENUM('clientInvoice', 'paymentReceipt', 'taxInvoice', 'otherIncome', 'socialInsurancePayment', 'housingFundPayment', 'salaryPayment', 'taxPayment', 'commercialInsurancePayment', 'benefitPayment', 'otherExpense'),
      allowNull: false
    },
    amount: {
      type: DataTypes.FLOAT,
      allowNull: false
    },
    date: {
      type: DataTypes.DATE,
      allowNull: false
    },
    description: DataTypes.TEXT,
    fileUrl: DataTypes.STRING
  });

  return FinancialDocument;
};

File: models/financial_document.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type FinancialDocument struct {
	gorm.Model
	DocumentType string    `json:"documentType" binding:"required"`
	ClientID     uint      `json:"clientId" binding:"required"`
	ContractID   uint      `json:"contractId" binding:"required"`
	EmployeeID   uint      `json:"employeeId"`
	Amount       float64   `json:"amount" binding:"required"`
	Date         time.Time `json:"date" binding:"required"`
	Description  string    `json:"description"`
	FileURL      string    `json:"fileUrl"`
}